SentryTestUtils/TestSentryNSTimerFactory.swift

```diff
@@ -1,52 +1,92 @@
 import Foundation
-import Sentry
+@testable import Sentry
 
 // We must not subclass NSTimer, see https://developer.apple.com/documentation/foundation/nstimer#1770465.
 // Therefore we return a NSTimer instance here with TimeInterval.infinity.
 public class TestSentryNSTimerFactory: SentryNSTimerFactory {
     public struct Overrides {
-    	private var _timer: Timer?
-    
-    	public var timer: Timer {
-        	get {
-         	   _timer ?? Timer()
```

Comment:
The fact that this returned a new timer if one had never been initialized I think was causing false negatives where tests should've been failing.